lib/tabry/cli/all_in_one.rb

```diff
@@ -83,8 +89,17 @@ def self.build(cli: nil, config: nil, &blk)
         # If we recognize command is going to be a completion command, fast track and
         # run completion now
         if run_completion
-          require_relative "../shells/bash/wrapper"
-          Tabry::Bash::Wrapper.run(ARGV[1], ARGV[2], config: config)
+          if ARGV.first == "completion"
+            fish_mode = ENV.fetch("TABRY_FISH_MODE", false)
+            if fish_mode
+              require_relative "../shells/fish/wrapper"
+              Tabry::Fish::Wrapper.run(ARGV[1], ARGV[2], config: config)
+            else
+              require_relative "../shells/bash/wrapper"
+              Tabry::Bash::Wrapper.run(ARGV[1], ARGV[2], config: config)
+            end
+            exit
```

Comment:
Added an early exit here, since this block is here to enable short-circuiting. Without it, I noticed the completions run twice and tabry takes twice as long to run..

Reply:
Oops. Good catch.